platform/osgi/platform-osgi-configadmin/src/main/java/org/apache/felix/cm/impl/ConfigurationManager.java

```diff
@@ -0,0 +1,1872 @@
+     *                   are to be augmented
+     * @param factoryPid the factory PID of the configuration object whose
+     *                   properties are to be augmented. This is non-<code>null</code>
+     *                   only for a factory configuration.
+     */
+    public void callPlugins(final Dictionary props, final ServiceReference sr,
+            final String configPid, final String factoryPid) {
+        ServiceReference<?>[] plugins = null;
+        try {
+            final String targetPid = (factoryPid == null) ? configPid : factoryPid;
+            String filter = "(|(!(cm.target=*))(cm.target=" + targetPid + "))";
+            plugins = bundleContext.getServiceReferences(ConfigurationPlugin.class.getName(),
+                    filter);
+        } catch (InvalidSyntaxException ise) {
+            // no filter, no exception ...
+        }
+
+        // abort early if there are no plugins
+        if (plugins == null || plugins.length == 0) {
+            return;
+        }
+
+        // sort the plugins by their service.cmRanking
+        if (plugins.length > 1) {
+            Arrays.sort(plugins, RankingComparator.CM_RANKING);
+        }
+
+        // call the plugins in order
+        for (int i = 0; i < plugins.length; i++) {
+            ServiceReference<?> pluginRef = plugins[i];
+            ConfigurationPlugin plugin = (ConfigurationPlugin) bundleContext.getService(pluginRef);
+            if (plugin != null) {
+                // if cmRanking is below 0 or above 1000, ignore modifications from the plugin
+                boolean ignore = false;
+                Object rankObj = pluginRef.getProperty(ConfigurationPlugin.CM_RANKING);
+                if (rankObj instanceof Integer) {
+                    final int ranking = ((Integer) rankObj).intValue();
+                    ignore = (ranking < 0) || (ranking > 1000);
+                }
+
+                try {
+                    plugin.modifyConfiguration(sr,
+                            ignore ? CaseInsensitiveDictionary.unmodifiable(props) : props);
+                } catch (Throwable t) {
+                    log(LogService.LOG_ERROR,
+                            "Unexpected problem calling configuration plugin {0}",
+                            new Object[] {toString(pluginRef), t});
+                } finally {
+                    // ensure ungetting the plugin
+                    bundleContext.ungetService(pluginRef);
+                }
+                ConfigurationImpl.setAutoProperties(props, configPid, factoryPid);
+            }
+        }
+    }
+
+    /**
+     * Creates a PID for the given factoryPid
+     *
+     * @param factoryPid
+     * @return
+     */
+    private static String createPid(String factoryPid) {
+        Random ng = numberGenerator;
+        if (ng == null) {
+            // FELIX-2771 Secure Random not available on Mika
+            try {
+                ng = new SecureRandom();
+            } catch (Throwable t) {
```

Comment:
❗️ Should we at least fix this catch(Throwable) and catch StackOverflowError|OutOfMemoryError|ThreadDeath and throw those back?

Reply:
As we just saw with the OOM discovered during HA testing, it is important. Let 's open a ticket to track this. We have a simple chance of at least handling 2 or 3 we know off right here so let's not continue with our eyes closed ;-)

Reply:
Eric is good with making these minor changes 😄

Reply:
Done 👍